### PR TITLE
4861 Removing invalid final access modifier

### LINKED ISSUE #4861: Error in Course / Speed Graph
Steps to reproduce:

1) Open a File , for example `sample.dpf`
2) Right click over the track, and the press "View XY Plot".
3) Choose course for example, and then zoom-in over a Graph generated.
4) Wait 1 or 2 minutes and the following error appears.

`com.thoughtworks.xstream.converters.reflection.ObjectAccessException: Invalid final field MWC.GUI.JFreeChart.ColouredDataItem._myColor
	at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.validateFieldAccess(PureJavaReflectionProvider.java:149)
	at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.visitSerializableFields(PureJavaReflectionProvider.java:105)
	at com.thoughtworks.xstream.converters.reflection.ReflectionConverter.marshal(ReflectionConverter.java:44)
	at com.thoughtworks.xstream.core.ReferenceByXPathMarshaller.convertAnother(ReferenceByXPathMarshaller.java:36)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeItem(AbstractCollectionConverter.java:68)
	at com.thoughtworks.xstream.converters.collections.CollectionConverter.marshal(CollectionConverter.java:47)
	at com.thoughtworks.xstream.core.ReferenceByXPathMarshaller.convertAnother(ReferenceByXPathMarshaller.java:36)
	at com.thoughtworks.xstream.converters.reflection.ReflectionConverter$1.writeField(ReflectionConverter.java:81)
	at com.thoughtworks.xstream.converters.reflection.ReflectionConverter$1.visit(ReflectionConverter.java:59)
	at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.visitSerializableFields(PureJavaReflectionProvider.java:108)
	at com.thoughtworks.xstream.converters.reflection.ReflectionConverter.marshal(ReflectionConverter.java:44)
	at com.thoughtworks.xstream.core.ReferenceByXPathMarshaller.convertAnother(ReferenceByXPathMarshaller.java:36)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeItem(AbstractCollectionConverter.java:68)
	at com.thoughtworks.xstream.converters.collections.CollectionConverter.marshal(CollectionConverter.java:47)
	at com.thoughtworks.xstream.core.ReferenceByXPathMarshaller.convertAnother(ReferenceByXPathMarshaller.java:36)
	at com.thoughtworks.xstream.converters.reflection.SerializableConverter$1.defaultWriteObject(SerializableConverter.java:156)
	at com.thoughtworks.xstream.converters.reflection.SerializableConverter.marshal(SerializableConverter.java:198)
	at com.thoughtworks.xstream.core.ReferenceByXPathMarshaller.convertAnother(ReferenceByXPathMarshaller.java:36)
	at com.thoughtworks.xstream.core.TreeMarshaller.start(TreeMarshaller.java:46)
	at com.thoughtworks.xstream.core.ReferenceByXPathMarshallingStrategy.marshal(ReferenceByXPathMarshallingStrategy.java:17)
	at com.thoughtworks.xstream.XStream.marshal(XStream.java:476)
	at com.thoughtworks.xstream.XStream.marshal(XStream.java:466)
	at com.thoughtworks.xstream.XStream.toXML(XStream.java:438)
	at org.mwc.cmap.xyplot.views.XYPlotView.saveState(XYPlotView.java:1247)
	at org.eclipse.ui.internal.ViewReference.persist(ViewReference.java:59)
	at org.eclipse.ui.internal.Workbench$13.run(Workbench.java:1225)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.ui.internal.Workbench.persist(Workbench.java:1215)
	at org.eclipse.ui.internal.Workbench.access$34(Workbench.java:1163)
	at org.eclipse.ui.internal.Workbench$40.runInUIThread(Workbench.java:2844)
	at org.eclipse.ui.progress.UIJob.lambda$0(UIJob.java:95)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:5113)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4666)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:635)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:559)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:400)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:660)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:597)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1468)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1441)`

### PR DESCRIPTION
**Pull request by saulhidalgoaular**
Fix #4861

### CODE REVIEW

**Comment by IanMayo** [comment]
This rings a bell. I think that "clean up" keeps making them final.  Are you able to add comments saying "do not make final, it will xstream serialisation", please?

I just looked if there was a special comment that would tell Eclipse not to clean-up that line, but they only exist for formatting.

**Comment by saulhidalgoaular** [comment]
> This rings a bell. I think that "clean up" keeps making them final. Are you able to add comments saying "do not make final, it will xstream serialisation", please?
> 
> I just looked if there was a special comment that would tell Eclipse not to clean-up that line, but they only exist for formatting.



> This rings a bell. I think that "clean up" keeps making them final. Are you able to add comments saying "do not make final, it will xstream serialisation", please?
> 
> I just looked if there was a special comment that would tell Eclipse not to clean-up that line, but they only exist for formatting.

oh, well spotted!!! I have fixed it.

File: org.mwc.cmap.legacy/src/MWC/GUI/JFreeChart/ColouredDataItem.java [hunk 1]
```diff
@@ -69,16 +69,21 @@
...
+	/**
+	 * WARNING. DON'T ADD THE FINAL MODIFIER TO THE FOLLOWING FIELDS.
+	 * https://github.com/debrief/debrief/issues/4861
+	 * Saul 
+	 */
 	/**
 	 * the color for this item
 	 */
-	private final Color _myColor;
+	private Color _myColor = null; // do not make final, it will xstream serialization
```
Comment (IanMayo) [inline org.mwc.cmap.legacy/src/MWC/GUI/JFreeChart/ColouredDataItem.java:80]: @saulhidalgoaular  - I think we need `break` between `will` and `xstream` :-)

Comment (saulhidalgoaular) [reply org.mwc.cmap.legacy/src/MWC/GUI/JFreeChart/ColouredDataItem.java]: Fixed :)

**Pull request merged**